apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -94,15 +100,32 @@ public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
         /// <inheritdoc />
         public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
         {
-            ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            return _client.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            if (_clientTask.IsCompleted)
+            {
+                ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                return _clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            }
+
+            if (Interlocked.Increment(ref _taskCounter) <= MaxWaitingTasks)
+            {
+                return _clientTask.ContinueWith(clientTask =>
+                {
+                    if (clientTask.IsFaulted)
+                    {
+                        throw new InvalidOperationException("The client task has faulted.");
+                    }
+                    ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                    return clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
```

Comment:
I believe you could leave out any options in that case - as far as I understand it, running them asynchronously is the default anyway... there may be some way of changing that default upstream, but I wouldn't worry about it. I think we'll be fine.